src/app/components/OfflineBanner/hook.ts

```diff
@@ -0,0 +1,48 @@
+import { Layer, useGetRuntimeStatus, useGetStatus } from '../../../oasis-indexer/api'
+import { Network } from '../../../types/network'
+import { SearchScope } from '../../../types/searchScope'
+import { AppError, AppErrors } from '../../../types/errors'
+import { useFormattedTimestampString } from '../../hooks/useFormattedTimestamp'
+import { paraTimesConfig } from '../../../config'
+
+export const useIsApiOffline = (network: Network): boolean => {
+  const query = useGetStatus(network)
+  return query.isFetched && !query.isSuccess
+}
+
+export type FreshnessInfo = {
+  outOfDate: boolean
+  lastUpdate?: string
+}
+
+export const useRuntimeFreshness = (scope: SearchScope): FreshnessInfo => {
+  const isApiOffline = useIsApiOffline(scope.network)
+  if (scope.layer === Layer.consensus) {
+    throw new AppError(AppErrors.UnsupportedLayer)
+  }
+  const query = useGetRuntimeStatus(scope.network, scope.layer)
+  const timeDistance = useFormattedTimestampString(query?.data?.data.latest_update)
+  if (isApiOffline || query.isLoading) {
+    return {
+      outOfDate: false,
```

Comment:
False if isApiOffline?

Reply:
Yes; if the API is offline, then we basically have no data. 

In this case I don't think it makes sense to add another banner, saying that the data is obsolete....

Reply:
Added comment as requested.